framework/src/outputs/TableOutput.C

```diff
@@ -34,6 +34,9 @@ InputParameters validParams<TableOutput>()
   InputParameters params = validParams<AdvancedOutput<FileOutput> >();
   params += AdvancedOutput<FileOutput>::enableOutputTypes("postprocessor scalar vector_postprocessor");
 
+  // Option for writing vector_postprocessor time file
+  params.addParam<bool>("time_data", false, "When true and VecptorPostprocessor data exists, write a csv file containing the timestep and time information.");
```

Comment:
@permcody You are almost correct, with one exception: the filename contains
the timestep (vpp_out_002, vpp_out_004, etc.). I included the timestep in
the _time file for convenience.

I did think about including the time and timestep in the file with a
something like this:

```
#time: 12345.6789
#timestep: 1342
x,y,z
1,2,3
```

The problem is that then the .csv file is not really a csv file, you have
to account for the commented meta data. So, either way something special
needs to happen when reading these files: account for a header or read a
second file. I went for the latter b/c the csv reader should produce csv
files, not some special format, in my opinion.


On Mon, Mar 28, 2016 at 10:50 AM, Cody Permann <notifications@github.com>
wrote:

> In framework/src/outputs/TableOutput.C
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6620-23discussion-5Fr57595158&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=Qc2LQQhMa_kHtvnXSAOlw_Z89b6djDNUIP2HRbZr-DU&s=XzsB2HxWWK9poKPxRVx08ZlFaq6e1BVgrAmnqhtOJoI&e=>
> :
>
> > @@ -34,6 +34,9 @@ InputParameters validParams<TableOutput>()
> >    InputParameters params = validParams<AdvancedOutput<FileOutput> >();
> >    params += AdvancedOutput<FileOutput>::enableOutputTypes("postprocessor scalar vector_postprocessor");
> >
> > +  // Option for writing vector_postprocessor time file
> > +  params.addParam<bool>("time_data", false, "When true and VecptorPostprocessor data exists, write a csv file containing the timestep and time information.");
>
> Let me comment to make sure I understand the design:
> Normally each vector postprocessor rights it's own CSV file, one row per
> timestep. Currently there is no time, or timestep information in that file.
>
> Question: I see two designs, either add this information to every vector
> postprocessor or create a separate file with that information. You've gone
> with the second and it does have the advantage that it does not duplicate
> information. It has the drawback that any tool that analyzes this
> information needs to open two files.
>
> Is that all correct?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6620_files_d54078de82adf10ec0e385253279dc0291999bfa-23r57595158&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=Qc2LQQhMa_kHtvnXSAOlw_Z89b6djDNUIP2HRbZr-DU&s=n766US5SywN7N0n-TbzOgw1g89b1zJQ-L9k-9E5jIDo&e=>
>